src/main/target/KIWIF4/target.h

```diff
@@ -17,25 +17,28 @@
 
 #pragma once
 
-#ifdef PLUMF4
+#define TARGET_CONFIG
+
+#if defined(PLUMF4)
 #define TARGET_BOARD_IDENTIFIER "PLUM"
 #define USBD_PRODUCT_STRING     "PLUMF4"
 
-#else
-	
+#elif defined(KIWIF4)
 #define TARGET_BOARD_IDENTIFIER "KIWI"
 #define USBD_PRODUCT_STRING     "KIWIF4"
 
+#elif defined(KIWIF4V2)	
```

Comment:
`KIWIF4` is defined for all variants, which means the `#elif defined(KIWIF4V2)` will never be taken. You have to reorder the conditions, putting `KIWIF4` last, preferably as `#else`.